jb/book_depth_statistics.hpp

```diff
@@ -0,0 +1,127 @@
+#ifndef jb_book_depth_stats_hpp
+#define jb_book_depth_stats_hpp
+
+#include <jb/config_object.hpp>
+#include <jb/event_rate_histogram.hpp>
+#include <jb/histogram.hpp>
+#include <jb/integer_range_binning.hpp>
+
+#include <iosfwd>
+
+namespace jb {
+
+/* Ticket https://github.com/coryan/jaybeams/issues/20
+ *
+ * Remove all the histograms copied from offline_statistics.*
+ * Create a new histogram to handle book depth samples
+ * Set the memory limit to 10000
+ *
+ */
+  
+/**
+ * Keep statistics about a feed and its offline processor.
+ *
+ * In JayBeams we will find ourselves processing all kinds of offline
+ * feeds, that is, files with the contents of a real-time feed.  We
+ * want to measure things like the processing time for the data, what
+ * is the distribution of message rates like, what is the distribution
+ * of mesage inter-arrival times like.
+ *
+ * This class encapsulates a lot of the logic for those computations.
+ * The class can be thoroughly configured at run-time, for ease of
+ * experimentation, but the default values should be reasonable.
+ *
+ * The source of timestamps can be an actual clock, the recorded
+ * timestamps is a feed file, or some virtual clock used in
+ * simulations.  The only requirement is for the timestamps to be
+ * compatible with std::chrono::duration<>.  It might seem odd to use
+ * std::chrono::duration<> instead of std::chrono::time_point<>, this
+ * is because most feeds are timestamps to a well known time point:
+ * midnight at the beginning of the day, or 00:00:00am.
+ *
+ * Likewise, the processing latency measurements can be generated from
+ * an actual clock or some virtual measurement (say counting
+ * interrupts, or CPU cycles vs. elapsed time).  The only requirement
+ * is for the measurements to be compatible with
+ * std::chono::duration<>.
+ *
+ * Note: In order to get the book_depth_t here jb/itch5/order_book_depth.hpp
+ * needs to be included. I decided to define book_depth_stats_t here to avoid that include
+ */
+typedef unsigned long int book_depth_stats_t;
+
+class book_depth_statistics {
+ public:
+  class config;
+
+  /// Constructor
+  explicit book_depth_statistics(config const& cfg);
+
+  /**
+   * Record a sample, that is book depth after the event
+   *
+   * @tparam event_timestamp_t the type used to record the event
+   * timestamps.
+   * @tparam book_depth_t the type used to record the book depth after processing the event
+   *
+   * @param ts the event timestamp, please see the class documentation
```

Comment:
Looks like the timestamp parameter can be deleted, not used, nor do I see how we would use it for book depth computations, please remove.